src/app.py

```diff
@@ -3,30 +3,67 @@
     Based on a user provided configuration, generate a set of random data
     pertaining to said configuration. Currently supported domain objects are:
 
-        * Back Office Position
-        * Cash Balance
-        * Cashflow
-        * Counterparty
-        * Depot Positions
-        * Front Office Position
         * Instrument
-        * Order Execution
+        * Account
+        * Trade
         * Price
-        * Stock Loan Position
-        * Swap Contract
-        * Swap Position
-
-    Each domain object to be generated (record count > 0) is generated in
-    turn to account for inter-object dependencies. Generation is
-    multiprocessed with a primary coordinator spawning two child process
-    coordinators. One of these children handles object generation, and the
-    second writing of these to file.
-
-    The generation coordinator received instruction to generate X number of
-    objects over Y number of subprocesses (Pool Size). Once these jobs are
-    executed and records returned, they are placed into a writing job queue.
-    The writing to file coordinator picks up & formats these tasks before
-    assigning them to execute on a second pool
+        * Front Office Position
+        * Back Office Position
+        * Depot Position
+        * Cash Balance
+        * Cash Flow
+        * Settlement Instruction
+
+    Domain objects are created sequentially, with one domain object having all
+    output files written before starting on the next domain object. This is
+    sequenced to ensure that inter-object dependencies are created correctly.
+
+    Dependencies are established using a local SQLite database file. When
+    creating a domain object, any fields that may be referenced by dependant
+    objects are persisted to the database. Upon creating a dependant object,
+    the database is queried to retrieve an appropriate value.
+
+    When creating domain object records and writing them to files,
+    multiprocessing is used to increase time efficiency. A Coordinator object
+    is responsible for creating and managing these processes, and it will
+    spawn two parent processes: 'create_parent_process' and
+    'write_parent_process'. These each spawn a number of child processes as
+    specified in the 'shared_args' section of the user config.
+
+    The Coordinator will populate a FIFO 'create_job_queue' with create jobs
+    (dictionaries specifying a quantity of domain object records to be created
+    and later written to output file). The maximum quantity of records in a
+    create job is specified in the 'shared_args' section of the user config.
+
+    The create parent process will wait until the create job queue is not
```

Comment:
Same as above.  I think it's best to use the actual name of any objects you're talking about